integration-tests/src/tests/nearcore/sync_nodes.rs

```diff
@@ -1,24 +1,147 @@
-use crate::genesis_helpers::genesis_block;
-use crate::nearcore_utils::{add_blocks, setup_configs};
-use crate::test_helpers::heavy_test;
-use actix::{Actor, System};
+use crate::tests::genesis_helpers::genesis_block;
+use crate::tests::test_helpers::heavy_test;
+use actix::{Actor, Addr, System};
 use futures::{future, FutureExt};
 use near_actix_test_utils::run_actix;
 use near_async::time::{Clock, Duration};
+use near_chain::Block;
 use near_chain_configs::test_utils::TESTING_INIT_STAKE;
 use near_chain_configs::Genesis;
-use near_client::{GetBlock, ProcessTxRequest};
+use near_client::{BlockResponse, ClientActor, GetBlock, ProcessTxRequest};
 use near_crypto::{InMemorySigner, KeyType};
 use near_network::tcp;
 use near_network::test_utils::{convert_boot_nodes, WaitOrTimeoutActor};
+use near_network::types::PeerInfo;
 use near_o11y::testonly::init_integration_logger;
 use near_o11y::WithSpanContextExt;
+use near_primitives::block::Approval;
+use near_primitives::hash::CryptoHash;
+use near_primitives::merkle::PartialMerkleTree;
+use near_primitives::num_rational::Ratio;
 use near_primitives::test_utils::create_test_signer;
 use near_primitives::transaction::SignedTransaction;
-use nearcore::{load_test_config, start_with_config};
+use near_primitives::types::validator_stake::ValidatorStake;
+use near_primitives::types::{BlockHeightDelta, EpochId};
+use near_primitives::validator_signer::ValidatorSigner;
+use near_primitives::version::PROTOCOL_VERSION;
+use nearcore::{load_test_config, start_with_config, NearConfig};
+use node_runtime::config::Rational32;
 use std::sync::atomic::{AtomicBool, Ordering};
 use std::sync::{Arc, RwLock};
 
+// This assumes that there is no height skipped. Otherwise epoch hash calculation will be wrong.
+pub fn add_blocks(
```

Comment:
These functions were the only use case of `integration-tests/src/nearcore_utils.rs` so moved these infile.